libcontainer/nsenter/nsexec.c

```diff
@@ -1223,8 +1224,14 @@ void nsexec(void)
 			 * was broken, so we'll just do it the long way anyway.
 			 */
 			write_log(DEBUG, "unshare remaining namespace (except cgroupns)");
-			if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) < 0)
-				bail("failed to unshare remaining namespaces (except cgroupns)");
+			for (; retries > 0; retries--) {
+				if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) >= 0)
+					break;
+				if (errno != EINVAL)
+					bail("failed to unshare remaining namespaces (except cgroupns)");
+				if (retries == 1)
+					bail("failed to unshare remaining namespaces (except cgroupns), please retry");
+			}
```

Comment:
Don't we need to add delay ?

Reply:
I think we don't nead to add delay. If we add a delay, the recurrence of the bug will affect the runc running speed. So I think we just need to try again quickly.